src/WaCore.Web/Middleware/SecureHeaders/WacSecureHeadersMiddleware.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using WaCore.Web.Middleware.SecureHeaders.Models;
+
+namespace WaCore.Web.Middleware.SecureHeaders
+{
+    /// <summary>
+    /// A middleware for injecting OWASP recommended headers into a
+    /// HTTP Request
+    /// </summary>
+    public class WacSecureHeadersMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly WaSecureHeadersMiddlewareConfiguration _config;
+
+        public WacSecureHeadersMiddleware(RequestDelegate next, WaSecureHeadersMiddlewareConfiguration config)
+        {
+            _next = next;
+            _config = config;
+        }
+
+        public void AddHeader(HttpContext httpContext, string headerName, string headerValue)
+        {
+            if (!httpContext.Response.Headers.ContainsKey(headerName))
+            {
+                httpContext.Response.Headers.Add(headerName, headerValue);
+            }
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+            if (_config == null)
+            {
+                throw new ArgumentException($@"Expected an instance of the
```

Comment:
ArgumentNullException would be suitable here if you moved the check to the constructor